rfc/cdevents-spinnaker.md

```diff
@@ -0,0 +1,106 @@
+# Implementing CDEvents to Spinnaker using Java SDK
+
+
+| | |
+|-|-|
+| **Status**     | _**Proposed**, Accepted, Implemented, Obsolete_ |
+| **RFC #**      | https://github.com/spinnaker/governance/pull/299 |
+| **Author(s)**  | _Jalander Ramagiri (`@rjalander`)_|
+| **SIG / WG**   | CDF-CDEvents, CDF-SIGEvents |
+
+## Overview
+
+Many organisations use more than one CI/CD tools as part of their pipelines and most of the times the organisations implement their own glue code in order to make the tools work with each other.
+
+Implementing [CDEvents](https://cdevents.dev/) to Spinnaker enables Spinnaker to communicate with other CI/CD tools with in the ecosystem. [CDEvents](https://cdevents.dev/) is a common specification for Continuous Delivery events based on CloudEvents.
+Spinnaker can produce and consume [CDEvents](https://cdevents.dev/) using [Java SDK](https://github.com/cdevents/sdk-java) which is developed and maintained by the [CDEvents community](https://cdevents.dev/community/).
+
+### Goals and Non-Goals
+Goals:
+- Event-Driven Standardization with CI/CD tools like Keptn, Tekton, Jenkins, Spinnaker, etc..
+- Create an `events-broker` that is able to produce and consume `CDEvents`
```

Comment:
Looked at this again, and mentioned it but sharing here:  Echo's implementation is PROBABLY the easiest path forward on this.  

For an example of a new Consumer: https://github.com/spinnaker/echo/tree/master/echo-pubsub-aws
https://github.com/spinnaker/echo/blob/master/echo-pubsub-aws/src/main/resources/s3.jinja for the jinja template used to do translation.  

I'd SUGGEST a new "notifications" type, aka "echo-notifications-queue" sub module (OR all of this as a plugin) to allow adding new notification to a different destination.  OR https://github.com/spinnaker/echo/tree/master/echo-notifications/src/main/groovy/com/netflix/spinnaker/echo add it here as a sub config as maybe a "queue" type... I could see some of that needing a better abstraction (aka the kork piece) for allowing different broker types.  Probably some more thinking along how this would best integrate...

Reply:
@jasonmcintosh it was nice meeting you in the Summit and discussing in-person about this integration.

One thing about the new consumer that you suggested to add like "echo-pubsub-knative".
But if we want not to have a provider specific implementation, do you suggest using an existing Webhook from Automated Trigger types or using an existing Pub/Sub System type(amazon/google) even for consuming a CDEvent message format (This we need to explore more on how to consume the CDEvent format using these types)

As we already discussed having Custom Webhook Stages for producing the CDEvent to a specific event-broker over HTTP POST or We can add new a notification type for producing CDEvent message to different brokers.

Please share your thoughts.